src/components/Table/Table.jsx

```diff
@@ -511,41 +479,194 @@ const Table = createClass({
 		className: string,
 
 		/**
-		 * Adjusts the style of the table to have more spacing within the table cells
+		 * Adjusts the row density of the table to have more or less spacing.
+		 */
+		density: oneOf(['compressed', 'extended']),
+
+		/**
+		 * Render the table with borders on the outer edge.
 		 */
-		hasExtraWhitespace: bool,
+		hasBorder: bool,
 
 		/**
-		 * render the table without borders on the outer edge
+		 * Enables word wrapping in tables cells.
 		 */
-		hasNoBorder: bool,
+		hasWordWrap: bool,
 	},
 
 	getDefaultProps() {
 		return {
-			hasNoBorder: false,
-			hasExtraWhitespace: false,
+			density: 'extended',
+			hasBorder: true,
+			hasWordWrap: true,
 		};
 	},
 
 	render() {
 
 		const {
 			className,
-			hasNoBorder,
-			hasExtraWhitespace,
+			hasBorder,
+			density,
+			hasWordWrap,
 		} = this.props;
 
 		return (
 			<table
 				{...this.props}
 				className={cx('&', {
-					'&-has-extra-whitespace': hasExtraWhitespace,
-					'&-has-no-border': hasNoBorder,
+					'&-density-extended': density === 'extended',
+					'&-density-compressed': density === 'compressed',
+					'&-has-border': hasBorder,
+					'&-has-word-wrap': hasWordWrap,
 				}, className)}
 			/>
 		);
 	}
 });
 
+/**
+ * mapToGrid
+ *
+ * Returns a 2 dimensional array of cell elements of the given component type. The map function can modify value of a cell.
+ */
+const mapToGrid = function mapToGrid(trList, cellType='td', mapFn=_.property('element')) {
+	const cellRowList = _.map(trList, (trElement) => _.map(filterTypes(trElement.props.children, cellType)));
+	const grid = [];
+
+	if (_.isEmpty(cellRowList)) {
+		return [];
+	}
+
+
+	// iterate over each row
+	for (let rowIndex=0; rowIndex<cellRowList.length; rowIndex++) {
```

Comment:
it was initially using both forEach and reduce but as i debugged issues with the first algorithm, it was just so much easier to debug this way, so i kept it.